lib/src/override.dart

```diff
@@ -0,0 +1,213 @@
+                        ? const WindowsOS()
+                        : id == 'android'
+                            ? const AndroidOS()
+                            : id == 'ios'
+                                ? const IOS()
+                                : id == 'fuchsia'
+                                    ? const FuchsiaOS()
+                                    : id == 'browser'
+                                        ? const BrowserOS()
+                                        : UnknownOS(id),
+            version);
+
+  /// Used by platforms which know the ID object.
+  const OperatingSystem._(this._osId, this.version);
+
+  /// Whether the operating system is a version of
+  /// [Linux](https://en.wikipedia.org/wiki/Linux).
+  ///
+  /// Identified by [id] being the string `linux`.
+  ///
+  /// This value is `false` if the operating system is a specialized
+  /// version of Linux that identifies itself by a different name,
+  /// for example Android (see [isAndroid]).
+  bool get isLinux => _osId is LinuxOS;
+
+  /// Whether the operating system is a version of
+  /// [macOS](https://en.wikipedia.org/wiki/MacOS).
+  ///
+  /// Identified by [id] being the string `macos`.
+  bool get isMacOS => _osId is MacOS;
+
+  /// Whether the operating system is a version of
+  /// [Microsoft Windows](https://en.wikipedia.org/wiki/Microsoft_Windows).
+  ///
+  /// Identified by [id] being the string `windows`.
+  bool get isWindows => _osId is WindowsOS;
+
+  /// Whether the operating system is a version of
+  /// [Android](https://en.wikipedia.org/wiki/Android_%28operating_system%29).
+  ///
+  /// Identified by [id] being the string `android`.
+  bool get isAndroid => _osId is AndroidOS;
+
+  /// Whether the operating system is a version of
+  /// [iOS](https://en.wikipedia.org/wiki/IOS).
+  ///
+  /// Identified by [id] being the string `ios`.
+  bool get isIOS => _osId is IOS;
+
+  /// Whether the operating system is a version of
+  /// [Fuchsia](https://en.wikipedia.org/wiki/Google_Fuchsia).
+  ///
+  /// Identified by [id] being the string `fuchsia`.
+  bool get isFuchsia => _osId is FuchsiaOS;
+
+  /// Whether running in a web browser.
+  ///
+  /// Identified by [id] being the string `browser`.
+  ///
+  /// If so, the [version] is the string made available
+  /// through `window.navigator.appVersion`.
+  bool get isBrowser => _osId is BrowserOS;
+}
+
+// Exposes the `OperatingSystem._` constructor to the conditionally imported
+// libraries.
+final class OperatingSystemInternal extends OperatingSystem {
+  const OperatingSystemInternal(super.id, super.version) : super._();
+}
+
+/// Operating identity object.
+///
+/// By only instantiating these subtypes guarded by target-OS guarded
+/// checks, unless using the "for testing" [OperatingSystem] constructor,
+/// all but one of the subclasses should be tree-shaken,
+/// and, e.g., the `_isId is IOS` test above should become tree-shakable
+/// on all other platforms.
+sealed class OS {
```

Comment:
It's not public as such, just shared with the cosmetically imported libraries.

But that also means it can easily be longer, without bothering anyone, like `RecognizedOperatingSystemKind`.

Reply:
I think it should be visible in the docs as a supertype - the subtypes of this class are public right?

Reply:
Yep, no subtype is public either. (Or shouldn't be, unless I made a mistake.)